package-res/resources/web/prompting/api/OperationAPI.js

```diff
@@ -48,22 +50,29 @@ define(['common-ui/prompting/PromptPanel', 'common-ui/prompting/parameters/Param
      * @method
      * @param {String} id HTML object id where the render the panel
      * @param {String} xml XML string containing the parameter definition
+     * @param {Function} getParameterXml The function called when the prompt panel needs to get the new parameter xml
      */
-    this.render = function(id, xml) {
+    this.render = function(id, xml, getParameterXml) {
```

Comment:
@krivera-pentaho being the devil's advocate, why do we need to pass an xml and a func to get an xml?
Can't we just pass the callback and we call it in the first loop?